### ISSUE TITLE
Action from AndroidPermission rationale is ignored

### ISSUE THREAD

**Issue by Ejdamm** [facebook/react-native#37493]
### Description

It's possible to use three different buttons in the dialog explaining why an Android permission is needed. See https://reactnative.dev/docs/permissionsandroid.html#request

The result from the dialog is never used.

### React Native Version

0.70.7

### Output of `npx react-native info`

System:
    OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (16) x64 AMD Ryzen 7 PRO 4750U with Radeon Graphics
    Memory: 3.98 GB / 14.86 GB
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 16.20.0 - ~/.nvm/versions/node/v16.20.0/bin/node
    Yarn: Not Found
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.0/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: javac 17 - /home/johndoe/.sdkman/candidates/java/current/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.7 => 0.70.7 
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

1. Request android permission with rationale
2. Press button
3. Don't get info about which button was clicked

### Snack, code example, screenshot, or link to a repository

https://github.com/facebook/react-native/blob/f10dd3f6660bbe6347b65e871dd11a2290426abe/packages/react-native/Libraries/PermissionsAndroid/PermissionsAndroid.js#LL260C14-L260C14

**Comment by phoneticallySAARTHaK** [comment]
> Forgot to search for closed issues. Duplicate of #28876

Is there any fix or workaround for this?

**Comment by Ejdamm** [comment]
> > Forgot to search for closed issues. Duplicate of #28876
> 
> Is there any fix or workaround for this?

Yes https://github.com/facebook/react-native/pull/28201#issuecomment-729580415
